File: platform/Java22/sweetline/src/main/java/com/qiplat/sweetline/HighlightConfig.java
package com.qiplat.sweetline;

/**
 * Highlight configuration
 *
 * @param showIndex   Whether the analysis result includes character index
 * @param inlineStyle Whether to use inline styles instead of returning style IDs
 * @param tabSize     Tab width used for indent guide level calculation
 */
public record HighlightConfig(boolean showIndex, boolean inlineStyle, int tabSize) {

    public HighlightConfig() {
        this(false, false, 4);
    }
}

File: platform/Apple/Sources/SweetLineCoreInternal/model/HighlightConfig.swift
import Foundation

public struct HighlightConfig: Sendable, Equatable {
    public let showIndex: Bool
    public let inlineStyle: Bool

    public init(showIndex: Bool = true, inlineStyle: Bool = false) {
        self.showIndex = showIndex
        self.inlineStyle = inlineStyle
    }

    public static let `default` = HighlightConfig()
}
